Source/UI/IdentityUIDelegate.swift

```diff
@@ -70,6 +85,21 @@ public protocol IdentityUIDelegate: class {
      - parameter done: call this to tell the flow to continue
      */
     func skipRequested(topViewController: UIViewController, done: @escaping (SkipLoginDisposition) -> Void)
+
+    /**
+     This will be called right before `didFinish` is called with a success result
+
+     You can use this to do other work to finish your login process. The `done` callback you call will tell the
+     UI what to do next.
+
+     The view controller passed in is whatever is being shown at the time. This can be nil in some cases where the
+     login flow did not need to fire up a flow (when validating a signing deep link when launching the app for e.g.)
+
+     - parameter result: the `User` that will be given to `didFinish`
+     - parameter topViewController: The currently shown top view controller
+     - parameter done: call this to say you're done so the UI can continue
+     */
+    func willSucceed(with user: User, on topViewController: UIViewController?, done: @escaping (LoginWillSucceedDisposition) -> Void)
```

Comment:
For some reason I expected this to be called `willFinish` since you talked about `didFinish` in the documentation.

Reply:
Hmm, I see I see.